invisible_cities/io/mcinfo_io.py

```diff
@@ -0,0 +1,311 @@
+
+import tables
+import numpy as np
+
+from .. reco import tbl_functions as tbl
+from .. core import system_of_units as units
+
+from .. evm.event_model import MCParticle
+from .. evm.event_model import MCHit
+from .. evm.event_model import Waveform
+
+from .. evm.nh5 import MCExtentInfo
+from .. evm.nh5 import MCHitInfo
+from .. evm.nh5 import MCParticleInfo
+
+from typing import Mapping
+
+# use Mapping (duck type) rather than dict
+
+
+class mc_info_writer:
+    """Write MC info to file."""
+    def __init__(self, h5file, compression = 'ZLIB4'):
+
+        self.h5file      = h5file
+        self.compression = compression
+        self._create_tables()
+        # last visited row
+        self.last_row = 0
+        self.last_written_hit = 0
+        self.last_written_particle = 0
+        self.first_extent_row = True
+
+    def _create_tables(self):
+        """Create tables in MC group in file h5file."""
+        if '/MC' in self.h5file:
+            MC = self.h5file.root.MC
+        else:
+            MC = self.h5file.create_group(self.h5file.root, "MC")
+
+        self.extent_table = self.h5file.create_table(MC, "extents",
+                                                     description = MCExtentInfo,
+                                                     title       = "extents",
+                                                     filters     = tbl.filters(self.compression))
+
+        # Mark column to index after populating table
+        self.extent_table.set_attr('columns_to_index', ['evt_number'])
+
+        self.hit_table = self.h5file.create_table(MC, "hits",
+                                                  description = MCHitInfo,
+                                                  title       = "hits",
+                                                  filters     = tbl.filters(self.compression))
+
+        self.particle_table = self.h5file.create_table(MC, "particles",
+                                                       description = MCParticleInfo,
+                                                       title       = "particles",
+                                                       filters     = tbl.filters(self.compression))
+
+    def __call__(self, mctables: (tables.Table, tables.Table, tables.Table),
+                 evt_number: int):
+
+        extents = mctables[0]
+        for iext in range(self.last_row, len(extents)):
+            this_row = extents[iext]
+            if this_row['evt_number'] < evt_number:
+                continue
+            if this_row['evt_number'] > evt_number:
+                break
```

Comment:
I think in this case, the `continue` and `break` statements are better read when put right after the colon, in the same line. Check it and if you like it commit the change.